starknet-foundry/crates/cast/src/starknet_commands/account/create.rs

```diff
@@ -0,0 +1,187 @@
+use crate::helpers::constants::OZ_CLASS_HASH;
+use crate::helpers::scarb_utils::get_property;
+use anyhow::{anyhow, Context, Result};
+use camino::Utf8PathBuf;
+use cast::{get_network, parse_number, print_formatted};
+use clap::Args;
+use rand::rngs::OsRng;
+use rand::RngCore;
+use scarb::ops::find_manifest_path;
+use serde_json::json;
+use starknet::accounts::{AccountFactory, OpenZeppelinAccountFactory};
+use starknet::core::types::FieldElement;
+use starknet::core::utils::get_contract_address;
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::JsonRpcClient;
+use starknet::signers::{LocalWallet, SigningKey};
+use std::fs::OpenOptions;
+use std::io::Write;
+use toml::Value;
+
+#[derive(Args, Debug)]
+#[command(about = "Create an account with all important secrets")]
+pub struct Create {
+    /// Account name under which secrets are going to be saved
+    #[clap(short, long)]
+    pub name: String,
+
+    /// Salt for the address
+    #[clap(short, long)]
+    pub salt: Option<FieldElement>,
+
+    // If passed, a profile with corresponding data will be created in Scarb.toml
+    #[clap(short, long)]
+    pub add_profile: bool,
+    // TODO: think about supporting different account providers
+}
+
+#[allow(clippy::too_many_arguments)]
+pub async fn create(
+    provider: &JsonRpcClient<HttpTransport>,
+    url: String,
+    accounts_file_path: Utf8PathBuf,
+    name: String,
+    network: &str,
+    maybe_salt: Option<FieldElement>,
+    add_profile: bool,
+) -> Result<Vec<(&'static str, String)>> {
+    let private_key = SigningKey::from_random();
+    let public_key = private_key.verifying_key();
+    let salt = match maybe_salt {
+        Some(salt) => salt,
+        None => FieldElement::from(OsRng.next_u64()),
+    };
+
+    let address = get_contract_address(
+        salt,
+        parse_number(OZ_CLASS_HASH)?,
+        &[public_key.scalar()],
+        FieldElement::ZERO,
+    );
+
+    let max_fee = {
+        let signer = LocalWallet::from_signing_key(private_key.clone());
+        let factory = OpenZeppelinAccountFactory::new(
+            parse_number(OZ_CLASS_HASH)?,
+            get_network(network)?.get_chain_id(),
+            signer,
+            provider,
+        )
+        .await?;
+        let deployment = factory.deploy(salt);
+
+        deployment.estimate_fee().await?.overall_fee
+    };
+
+    if !accounts_file_path.exists() {
```

Comment:
What is the point of this? Maybe we should break immediately? From what I understand this is going to break anyway since there will be no accounts and a desired account will not be found, right?

Reply:
As you can see we check if accounts file exists and if not we just create it

Reply:
Yes but what is the point of creating this file? It's not intuitive, because a user gets something like this: `account xxx does not exist` with no clue it comes from a certain file, right? We could print an error like `accounts.json does not exist, create one and put accounts there ...`.

Reply:
Ok, this error is fine but how is it connected with this PR? This file is created because someone is creating an account (so its secrets have to be saved somewhere)